cmd/eksctl-anywhere/cmd/import_images.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
+
+*/
+package cmd
+
+import (
+	"context"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/eks-anywhere/cmd/eksctl-anywhere/cmd/internal/commands/artifacts"
+	"github.com/aws/eks-anywhere/pkg/bundles"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/docker"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/helm"
+	"github.com/aws/eks-anywhere/pkg/tar"
+)
+
+// imagesCmd represents the images command
+var importImagesCmd = &cobra.Command{
+	Use:   "images",
+	Short: "Import images and charts to a registry from a tarball",
+	Long: `Import all the images and helm charts necessary for EKS Anywhere clusters into a registry.
+Use this command in conjunction with download images, passing it output tarball as input to this command.`,
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
+		return importImagesCommand.Call(ctx)
+	},
+}
+
+func init() {
+	importCmd.AddCommand(importImagesCmd)
+
+	importImagesCmd.Flags().StringVarP(&importImagesCommand.InputFile, "input", "i", "", "Input tarball containing all images and charts to import")
+	if err := importImagesCmd.MarkFlagRequired("input"); err != nil {
+		log.Fatalf("Cannot mark 'input' as required: %s", err)
+	}
+	importImagesCmd.Flags().StringVarP(&importImagesCommand.RegistryEndpoint, "registry", "r", "", "Registry where to import images and charts")
+	if err := importImagesCmd.MarkFlagRequired("registry"); err != nil {
+		log.Fatalf("Cannot mark 'registry' as required: %s", err)
+	}
+	importImagesCmd.Flags().StringVarP(&importImagesCommand.BundlesFile, "bundles", "b", "", "Bundles file to read artifact dependencies from")
+	if err := importImagesCmd.MarkFlagRequired("bundles"); err != nil {
+		log.Fatalf("Cannot mark 'bundles' as required: %s", err)
+	}
+}
+
+var importImagesCommand = ImportImagesCommand{}
+
+type ImportImagesCommand struct {
+	InputFile        string
+	RegistryEndpoint string
+	BundlesFile      string
+}
+
+func (c ImportImagesCommand) Call(ctx context.Context) error {
+	factory := dependencies.NewFactory()
+	deps, err := factory.
+		WithManifestReader().
+		Build(ctx)
+	if err != nil {
+		return err
+	}
+
+	bundle, err := bundles.Read(deps.ManifestReader, c.BundlesFile)
+	if err != nil {
+		return err
+	}
+
+	artifactsFolder := "tmp-eks-a-artifacts"
+	dockerClient := executables.BuildDockerExecutable()
+	toolsImageFile := filepath.Join(artifactsFolder, eksaToolsImageTarFile)
+
+	// Import the eksa tools image into the registry first, so it can be used immediately
+	// after to build the helm executable
+	importToolsImage := artifacts.ImportToolsImage{
+		Bundles:            bundle,
+		InputFile:          c.InputFile,
+		TmpArtifactsFolder: artifactsFolder,
+		UnPackager:         tar.NewPackager(),
```

Comment:
Packager is the struct that is able to package and unpackage
But this command receives a UnPackager entity, because it only needs to unpackage. Imo it's ok to have on struct that implements multiple interfaces, but I think entities should accept only the smaller interface they need (in general)